stix_shifter/stix_transmission/src/modules/utils/RestApiClient.py

```diff
@@ -6,31 +6,61 @@
 import errno
 
 # This is a simple HTTP client that can be used to access the REST API
-class RestApiClient:
 
-    def __init__(self, host, port = None, cert=None, headers={}, url_modifier_function=None, cert_verify=True):
+
+class RestApiClient:
+    #cert_verify can be True, False, or a Cert
+    def __init__(self, host, port=None, cert=None, headers={}, url_modifier_function=None, cert_verify=True):
         server_ip = host
         if port is not None:
-            server_ip += ":" + str(port)            
+            server_ip += ":" + str(port)
         self.server_ip = server_ip
-        self.cert_verify = str(cert_verify).lower() not in ['0', 'f', 'false', 'f', 'n', 'no', 'disable', 'disabled']
-        self.cert = cert
+
+        if isinstance(cert_verify, bool):
+            self.server_cert_content = None
+            self.server_cert_exist = False
+            self.cert_verify = cert_verify
+        elif isinstance(cert_verify, str):
+            self.server_cert_content = cert_verify
+            self.server_cert_exist = True
+            self.cert_verify = None
+
+        if cert is not None:
+            self.client_cert_exist = True
+        else:
+            self.client_cert_exist = False
+
+        self.client_cert_content = cert
+        self.client_cert = "client_cert.pem"
+        self.server_cert = "server_cert.pem"
+
         self.headers = headers
         self.url_modifier_function = url_modifier_function
 
-
     # This method is used to set up an HTTP request and send it to the server
     def call_api(self, endpoint, method, headers=None, params=[], data=None, urldata=None):
-        self.cert_file_name = None
+        #convert client cert to file
         try:
-            if self.cert is not None and self.cert_verify:
-                # put key/cert pair into a file to read it later
-                self.cert_file_name = "cert.pem"
-                with open(self.cert_file_name, 'w') as f:
+            if self.client_cert_content is not None:
+                with open(self.client_cert, 'w') as f:
+                    try:
+                        f.write(self.client_cert_content)
+                    except IOError:
+                        print('Failed to setup certificate')
+                self.client_cert_content = self.client_cert
+            else:
+                self.client_cert_content = None
+
+            # covnert server cert to file
+            if self.server_cert_content is not None:
+                with open(self.server_cert, 'w') as f:
                     try:
-                        f.write(self.cert)
+                        f.write(self.server_cert_content)
                     except IOError:
                         print('Failed to setup certificate')
+                self.server_cert_content = self.server_cert
+            else:
+                self.server_cert_content = None
```

Comment:
Isn't `self.server_cert_content` already `None`?

Reply:
Yea Azam is going over the logic I wrote it fast just so we could get it  into a PR